pyiron_base/generic/datacontainer.py

```diff
@@ -510,6 +510,43 @@ def get(self, key, default=None, create=False):
         else:
             return super().get(key, default=default)
 
+    def search_tree(self, key, fail=True):
+        """
+        Search for ``key`` in the Container hierarchy.
+
+        This should be used if there is only one item in the hierarchy.
+        If the key appears multiple times, there is no control over which
+        item is found.
+
+        Args:
+            key (str):          the key to look for
+            fail (bool):        what to do if key is not found.
+                                True  => raise IndexError
+                                False => return None
+        Raise:
+            TypeError:  if key is not str
+            IndexError: if key is not found and fail is True.
+
+        Returns:
+            object: element at ``key``
+        """
+
+        if not isinstance (key,str):
+            raise TypeError ("Cannot search for non-string key.")
+
+        # search within current level
+        if (key in self): return self.get(key)
+
+        # descend into subgroups
+        for it in self.groups ():
+            pick = search_tree(self[it], key, False)
```

Comment:
Thinking about it again, I guess for the small containers depth-breadth first doesn't make much of a performance difference, so depth-first is fine with me.

Reply:
```suggestion
            pick = self[it].search_tree(key, False)
```
;P

Reply:
Agreed. I had developed this outside the class and forgot to do it right.